extract.js

```diff
@@ -34,15 +35,15 @@ function extract (spec, dest, opts) {
       }
     })
   } else {
-    opts.log.silly('pacote', 'no tarball hash provided for', spec, '- extracting by manifest')
+    opts.log.silly('pacote', 'no tarball hash provided for', spec.name, '- extracting by manifest')
     return retry((tryAgain, attemptNum) => {
       return extractByManifest(
         startTime, spec, dest, opts
       ).catch(err => {
         // We're only going to retry at this level if the local cache might
         // have gotten corrupted.
         if (err.code === 'EINTEGRITY' && opts.cache) {
-          opts.log.warn('pacote', `tarball integrity check for ${spec} failed. Clearing cache entry. ${err.message}`)
+          opts.log.warn('pacote', `tarball integrity check for ${spec.name}@${spec.saveSpec || spec.fetchSpec} failed. Clearing cache entry. ${err.message}`)
```

Comment:
Do you want me to add a `toString` method on spec objects to make debugging them easier?